### ISSUE TITLE
Change viewBox and width/height from lottie-react?

### ISSUE THREAD

**Issue by ivanjeremic** [Gamote/lottie-react#20]
Is there any lottie-react way of changing the svg viewBox and width/height?

Or do I need to get the element by tag and change it the non react way?

The reason I need to change this I want my svg to be 100vh on every device but somehow with the lottie styles only this is not possible, but I would be happy about any suggestion. Thanks

**Comment by Gamote** [comment]
@ivanjeremic That should be the default behaviour of Lottie. Please create an example in [CodeSandbox](https://codesandbox.io) and I will try to come up with a solution starting from your current state.

**Comment by ivanjeremic** [comment]
> @ivanjeremic That should be the default behaviour of Lottie. Please create an example in [CodeSandbox](https://codesandbox.io) and I will try to come up with a solution starting from your current state.

Thanks, Editing the JSON helped me solving this, because I get the JSON from a designer and I have no access to the build process of the animation which is also the reason to find a way for element interaction, so some Circle that looks like a button in the svg which I can target from inside React for button clicks.

What would be the lottie-react way of doing this? Does the designer need to add with the AfterEffects lottie framework these circles and assign IDs to them so I can target them with JS? Or maybe even better target them the React way? sadly this is something that can't be shown in a codesandbox I only need to know the process how to get clickable elements in lottie-react and what exactly the designer needs to look for?  Thanks

**Comment by Gamote** [comment]
@ivanjeremic While there's no direct/predefined way to do this in `lottie-react` I think you can achieve this by using the `useLottie` hook to get access to the animation instance. From there you have full access to the Lottie object which could give you some methods that may be helpful (e.g. `renderer`).

Search for similar questions to [this](https://stackoverflow.com/questions/51015882/can-i-add-interactivity-js-css-to-affect-elements-within-bodymovin-lottie).

**Comment by ivanjeremic** [comment]
> @ivanjeremic While there's no direct/predefined way to do this in `lottie-react` I think you can achieve this by using the `useLottie` hook to get access to the animation instance. From there you have full access to the Lottie object which could give you some methods that may be helpful (e.g. `renderer`).
> 
> Search for similar questions to [this](https://stackoverflow.com/questions/51015882/can-i-add-interactivity-js-css-to-affect-elements-within-bodymovin-lottie).

The Stackoverflow helped me to find the `renderer` in `animationItem` but still the `DOMLoaded` eventListener does not work it is undefined even if I put it in a `useEffect` and there is no description in the Docs how to use `DOMLoaded` correctly in react maybe you have a hint for me? Thanks

**Comment by Gamote** [comment]
@ivanjeremic For me `DOMLoaded` event is working just fine: https://codesandbox.io/s/lottie-react-hook-forked-c0fed. Regarding the other issue, **please** recreate it in **CodeSandbox** so I have a concrete example.

**Comment by ivanjeremic** [comment]
> @ivanjeremic For me `DOMLoaded` event is working just fine: https://codesandbox.io/s/lottie-react-hook-forked-c0fed. Regarding the other issue, **please** recreate it in **CodeSandbox** so I have a concrete example.

I edited my issue hope you saw the edit section, but regardless I will tomorrow create a codesandbox for you so you can see it.

**Comment by ivanjeremic** [comment]
The CodeSandbox was easier to make then I thought, I used the example for element interaction from your StackOverflow link and combined it with a fork of your CodeSandBox [https://codesandbox.io/s/lottie-react-hook-renderer-problem-qv4lt?file=/src/App.js](https://codesandbox.io/s/lottie-react-hook-renderer-problem-qv4lt?file=/src/App.js)

**Comment by Gamote** [comment]
You get undefined because you've used `lottieAnimation` instead of `animationItem`: [CodeSandbox](https://codesandbox.io/s/lottie-react-hook-forked-c0fed?file=/src/App.js)

**Comment by ivanjeremic** [comment]
> You get undefined because you've used `lottieAnimation` instead of `animationItem`: [CodeSandbox](https://codesandbox.io/s/lottie-react-hook-forked-c0fed?file=/src/App.js)

Thanks really my mistake, the API is so hard to parse, but `baseElement` does not react to clicks do I miss here something?

**Comment by Gamote** [comment]
I don't know to be honest because I didn't explored this use case. I will recommend you to ask on `lottie-web` repo as you have access to the Lottie instance or to create a future request so we don't go to much of track.

Currently I'm working on `v3` so the response may be slow, but I can assure you that I will look into this for the new version.

**Comment by ivanjeremic** [comment]
> I don't know to be honest because I didn't explored this use case. I will recommend you to ask on `lottie-web` repo as you have access to the Lottie instance or to create a future request so we don't go to much of track.
> 
> Currently I'm working on `v3` so the response may be slow, but I can assure you that I will look into this for the new version.

Thanks hope V3 will have more APIs for the React usage like some hooks for click events for specific Elements
